src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -250,15 +249,10 @@
         "%", SymbolSegment, name="modulo", type="binary_operator"
     ),
     SlashSegment=StringParser("/", SymbolSegment, name="slash", type="slash"),
-    ConcatSegment=StringParser(
-        "||", SymbolSegment, name="concatenate", type="binary_operator"
-    ),
-    BitwiseAndSegment=StringParser(
-        "&", SymbolSegment, name="binary_and", type="binary_operator"
-    ),
-    BitwiseOrSegment=StringParser(
-        "|", SymbolSegment, name="binary_or", type="binary_operator"
+    AmpersandSegment=StringParser(
+        "&", SymbolSegment, name="ampersand", type="ampersand"
```

Comment:
You can have both `BitwiseAndSegment` and `AmpersandSegment` defined here, saving you having to redefine it as a class later.